compose/config/config.py

```diff
@@ -874,6 +874,20 @@ def finalize_service_volumes(service_dict, environment):
     return service_dict
 
 
+def finalize_service_secrets(service_dict, environment):
+    if 'secrets' in service_dict:
+        finalized_secrets = []
+        normalize = environment.get_boolean('COMPOSE_CONVERT_WINDOWS_PATHS')
+        win_host = environment.get_boolean('COMPOSE_FORCE_WINDOWS_HOST')
+        for v in service_dict['secrets']:
+            if isinstance(v, dict):
```

Comment:
I'm unsure about this check, at this point I think what we must have in our hand is the result of an earlier call to `types.ServiceSecret.parse(s)` which is a subclass of `collections.namedtuple` with a `repr` that returns a `dict`.

So I think this if can be removed, i.e .making the `finalized_secrets.append(...)` unconditional.

The variant in `finalize_service_volumes` is different because in that context the thing in hand could be a `dict` (`MountSpec`) or a `tuple` (`VolumeSpec`) hence the check.